spec/03-plugins/10-key-auth/01-api_spec.lua

```diff
@@ -310,4 +311,117 @@ describe("Plugin: key-auth (API)", function()
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(6, #json.data)
+        assert.equal(6, json.total)
+      end)
+      it("retrieve key-auths for a consumer_id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/key-auth?consumer_id=" .. consumer.id
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(3, #json.data)
+        assert.equal(3, json.total)
+      end)
+      it("retrieve key-auths for a consumer_id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/key-auth",
+          body = {
+            consumer_id = consumer.id
+          },
+          headers = {
+            ["Content-Type"] = "application/json"
+          }
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(3, #json.data)
+        assert.equal(3, json.total)
+      end)
+      it("return empty for a non-existing consumer_id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/key-auth",
+          body = {
+            consumer_id = utils.uuid()
+          },
+          headers = {
+            ["Content-Type"] = "application/json"
+          }
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(0, #json.data)
+        assert.equal(0, json.total)
+      end)
+    end)
+  end)
+  describe("/key-auth/:credential_key_or_id/consumer", function()
+    describe("GET", function()
+      local credential;
+      setup(function()
+        helpers.dao:truncate_table("keyauth_credentials")
+        credential = assert(helpers.dao.keyauth_credentials:insert {
+          consumer_id = consumer.id
+        })
+      end)
+      it("retrieve consumer from a key id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/key-auth/" .. credential.id .. "/consumer"
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.same(consumer,json)
+      end)
+      it("retrieve consumer from a keyid", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/key-auth/" .. credential.key .. "/consumer"
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.same(consumer,json)
+      end)
+      it("returns 404 for a random id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/key-auth/" .. utils.uuid()  .. "/consumer"
```

Comment:
Could we also add a test case with some non-UUID garbage? The reason is that if the value is not recognized as a valid UUID, then the admin API will treat it as the credential key. We need to make sure a non-existing key also returns HTTP 404. Thanks!